src/execution/operator/join/physical_range_join.cpp

```diff
@@ -227,11 +231,16 @@ idx_t PhysicalRangeJoin::LocalSortedTable::MergeNulls(const vector<JoinCondition
 		}
 	}
 
-	auto &primary = keys.data[0];
 	if (all_constant == keys.data.size()) {
 		//	Either all NULL or no NULLs
+		if (ConstantVector::IsNull(primary)) {
+			// Primary is already NULL
+			return count;
+		}
 		for (auto &v : keys.data) {
 			if (ConstantVector::IsNull(v)) {
+				// Create a new validity mask to avoid modifying original mask
+				ConstantVector::Validity(primary).Initialize(nullptr);
```

Comment:
This still modifies the original `ConstantVector::Validity(Vector &)` returns a `ValidityMask &` so calling `Initialize` will modify it.

Reply:
It doesn't modify the original mask since `primary` is a copy of `keys.data[0]` rather than its reference. I'm not sure if there is a more readable way to do this.

Reply:
Ah but the assignment at line 46 doesn't copy the internal data - it just copies pointers to it. Just use the same `pvalidity_copy` approach you used in the non-constant case.

Reply:
OK I've fixed it.